projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -31,201 +25,231 @@ export class SelectPositioningStrategy extends ConnectedPositioningStrategy impl
-                        return;
-                    }
-                    if (OUT_OF_BOUNDS.Direction === Direction.Bottom) {
-                        // (more than 5 items) and no current scroll
-                        if (currentScroll === 0) {
-                            this.positionNoScroll(contentElement, CURRENT_POSITION_Y);
-                            return;
-                            // (more than 5 items) and current scroll
-                        } else {
-                            this.positionAndScrollBottom(contentElement, OUT_OF_BOUNDS.Amount);
-                            return;
-                        }
-                    }
-                }
-                // if there is no enough remaining scroll to scroll the item
-                if (remainingScroll < itemHeight) {
-                    if (OUT_OF_BOUNDS.Direction === Direction.Top) {
-                        this.positionNoScroll(contentElement, CURRENT_POSITION_Y);
-
-                    }
-                    if (OUT_OF_BOUNDS.Direction === Direction.Bottom) {
-                        this.positionAndScrollBottom(contentElement, OUT_OF_BOUNDS.Amount);
-                    }
-                }
-            }
-            // (more than 5 items) and no remaining scroll
-            if (remainingScroll === 0 && currentScroll !== 0) {
-                if (OUT_OF_BOUNDS) {
-                    if (OUT_OF_BOUNDS.Direction === Direction.Bottom) {
-                        this.positionAndScrollBottom(contentElement, OUT_OF_BOUNDS.Amount);
-                        return;
-                    }
-                }
-                this.positionNoScroll(contentElement, CURRENT_POSITION_Y);
-            }
-        }
+        return itemElement;
+    }
+
+    protected calculateYoffset(selectFit: SelectFit) {
+        const contentElementTopLeftPointY = selectFit.contentElementRect.top;
+        selectFit.verticalOffset =
+            -(selectFit.itemRect.top - contentElementTopLeftPointY + selectFit.styles.itemTextToInputTextDiff);
+        this.global_yOffset = selectFit.verticalOffset;
     }
+
+    protected calculateXoffset(selectFit: SelectFit) {
+        selectFit.horizontalOffset = selectFit.styles.itemTextIndent - selectFit.styles.itemTextPadding;
+        this.global_xOffset = selectFit.horizontalOffset;
+    }
+}
+
+export interface SelectFit extends ConnectedFit {
+    contentElementRect?: ClientRect;
+    targetRect?: ClientRect;
+    viewPortRect?: ClientRect;
+    fitHorizontal?: boolean;
+    fitVertical?: boolean;
+    left?: number;
+    right?: number;
+    top?: number;
+    bottom?: number;
+    horizontalOffset?: number;
+    verticalOffset?: number;
+
+    // New properties
+    inputElement?: HTMLElement;
+    dropDownList?: HTMLElement;
+    itemElement?: HTMLElement;
+    itemRect?: ClientRect;
+    styles?: SelectStyles;
 }
+
+export interface SelectStyles {
+    itemTextPadding?: number;
+    itemTextIndent?: number;
```

Comment:
You are using `itemTextPadding` and `itemTextIndent` only once in `calculateXoffset`. Consider removing these two properties from `SelectStyles` and move their calculations in `calculateXoffset`

Reply:
it may be so, and yet I prefer to store the style-related variables in one place - SelectStyles obj.

Reply:
this is TODO :)